internal/cmd/local/docker/docker.go

```diff
@@ -153,3 +175,205 @@ func (d *Docker) Port(ctx context.Context, container string) (int, error) {
+		return errors.New(fmt.Sprintf("volume %s does not exist", volume))
+	}
+
+	// create a container for running the `docker cp` command
+	// docker run -d -v airbyte_db:/var/lib/postgresql/data alpine:3.20 tail -f /dev/null
+	conCopy, err := d.Client.ContainerCreate(
+		ctx,
+		&container.Config{
+			Image:      "alpine:3.20",
+			Entrypoint: []string{"tail", "-f", "/dev/null"},
+		},
+		&container.HostConfig{
+			Mounts: []mount.Mount{{
+				Type:   mount.TypeVolume,
+				Source: volume,
+				Target: migratePGDATA,
+			}},
+		},
+		nil,
+		nil,
+		"")
+	if err != nil {
+		return fmt.Errorf("could not create initial docker migrate container: %w", err)
+	}
+	pterm.Debug.Println(fmt.Sprintf("Created initial migration container '%s'", conCopy.ID))
+
+	// docker cp [conCopy.ID]]:/$migratePGDATA/. ~/.airbyte/abctl/data/airbyte-volume-db/pgdata
+	userHome, _ := os.UserHomeDir()
+	dst := filepath.Join(userHome, ".airbyte", "abctl", "data", "airbyte-volume-db", "pgdata")
+	// note the src must end with a `.`, due to how docker cp works with directories
+	if err := d.copyFromContainer(ctx, conCopy.ID, migratePGDATA+"/.", dst); err != nil {
+		return fmt.Errorf("could not copy airbyte db data from container %s: %w", conCopy.ID, err)
+	}
+	pterm.Debug.Println(fmt.Sprintf("Copied airbyte db data from container '%s' to '%s'", conCopy.ID, dst))
+
+	d.stopContainer(ctx, conCopy.ID)
+
+	// Create a container for adding the correct db user and renaming the database.
+	// We have inconsistencies between our docker and helm default database credentials and even our database name.
+	// docker run
+	// -e POSTGRES_USER=docker -e POSTGRES_PASSWORD=docker -e POSTGRES_DB=airbyte -e PGDATA=/var/lib/postgresql/data \
+	// -v ~/.airbyte/abctl/data/airbyte-volume-db/pgdata:/var/lib/postgresql/data
+	// postgres:13-alpine
+	conTransform, err := d.Client.ContainerCreate(
+		ctx,
+		&container.Config{
+			Image: "postgres:13-alpine",
+			Env: []string{
+				"POSTGRES_USER=docker",
+				"POSTGRES_PASSWORD=docker",
+				"POSTGRES_DB=postgres",
+				"PGDATA=" + migratePGDATA,
+			},
+		},
+		&container.HostConfig{
+			Mounts: []mount.Mount{{
+				Type:   mount.TypeBind,
+				Source: dst,
+				Target: migratePGDATA,
+			}},
+		},
+		nil,
+		nil,
+		"")
+	if err != nil {
+		return fmt.Errorf("could not create docker container: %w", err)
+	}
+	pterm.Debug.Println(fmt.Sprintf("Created secondary migration container '%s'", conTransform.ID))
+	pterm.Debug.Println(fmt.Sprintf("Container was created with the following warnings: %s", conTransform.Warnings))
+	if err := d.Client.ContainerStart(ctx, conTransform.ID, container.StartOptions{}); err != nil {
+		return fmt.Errorf("could not start container %s: %w", conTransform.ID, err)
+	}
+	// cleanup and remove container when we're done
+	defer func() { d.stopContainer(ctx, conTransform.ID) }()
+
+	// TODO figure out a better way to determine when the container has successfully started
+	time.Sleep(10 * time.Second)
```

Comment:
I'm concerned that with slower internet connections, this might not be enough time. Can we write a simple `waitForStatus(id, status, timeout)` helper that could be used here? and further down. It looks like below there is an implementation of just such a loop in `exec()`

Reply:
I'm going to handle this in a follow-up PR. Get this beast of a PR merged in first.